File: examples/python/uppercase_node.py
"""A complete synchronous Muxiva Python Node."""

import muxiva


class UppercaseNode(muxiva.TransformNode):
    def on_process(self, frame: muxiva.TextFrame):
        return muxiva.TextFrame(
            frame.text.upper(),
            timestamp_ns=frame.timestamp_ns,
            sequence=frame.sequence,
        )


with muxiva.NodeRunner(UppercaseNode()) as runner:
    [output] = runner.process(muxiva.TextFrame("hello muxiva", sequence=1))
    assert output.text == "HELLO MUXIVA"
    print(output.text)

File: examples/typescript/uppercase-node.ts
import { NodeRunner, defineTransformNode } from '@muxiva/core'

type TextValue = { kind: 'text'; text: string; sequence: number }

const uppercase = defineTransformNode<TextValue, TextValue>({
  onProcess(frame) {
    return { ...frame, text: frame.text.toUpperCase() }
  },
})

const runner = new NodeRunner(uppercase)
const output = await runner.process({ kind: 'text', text: 'hello muxiva', sequence: 1 })
if (Array.isArray(output) || output == null || output.text !== 'HELLO MUXIVA') {
  throw new Error('unexpected Node output')
}
console.log(output.text)
await runner.finish()
await runner.close()
